File: jruby/identify.rb
#!/bin/env jruby

include Java

# ============================================================
#   Imports
# ============================================================

import org.gavrog.joss.pgraphs.io.Net
import org.gavrog.joss.pgraphs.io.Archive

import java.lang.ClassLoader
import java.io.InputStreamReader
import java.io.BufferedReader

# ============================================================
#   Prepare for RCSR and Zeolite Atlas lookup
# ============================================================

def archive_read(archive, path)
    # --- make sure this works from within .jar files and such
    stream = ClassLoader.getSystemResourceAsStream(path)
    reader = BufferedReader.new(InputStreamReader.new(stream))
    archive.add_all reader
end

# --- create an empty archive
archive = Archive.new "1.0"

# --- add entries from RCSR and zeolite archive files
archive_read archive, "org/gavrog/apps/systre/rcsr.arc"
archive_read archive, "org/gavrog/apps/systre/zeolites.arc"


# ============================================================
#   Main loop: read nets and print their symbols if found
# ============================================================

Net.iterator(ARGV[0]).each do |net|
    if not net.locally_stable?
        puts ">>>unstable<<<"
    elsif net.ladder?
    	puts ">>>ladder<<<"
    elsif found = archive.get(net.minimal_image.systre_key)
        puts found.name
    else
        puts ">>>unknown<<<"
    end
end

# ============================================================
#   EOF
# ============================================================

File: jython/identify.py
#!/bin/env jython

# ============================================================
#   Global imports
# ============================================================

import sys

from org.gavrog.joss.pgraphs.io import Net
from org.gavrog.joss.pgraphs.io import Archive


# ============================================================
#   Prepare for RCSR and Zeolite Atlas lookup
# ============================================================

def archive_read(archive, path):
    from java.lang import ClassLoader
    from java.io import InputStreamReader, BufferedReader

    # --- make sure this works from within .jar files and such
    stream = ClassLoader.getSystemResourceAsStream(path)
    reader = BufferedReader(InputStreamReader(stream))
    archive.addAll(reader)

# --- create an empty archive
archive = Archive("1.0")

# --- add entries from RCSR and zeolite archive files
archive_read(archive, "org/gavrog/apps/systre/rcsr.arc")
archive_read(archive, "org/gavrog/apps/systre/zeolites.arc")


# ============================================================
#   Main loop: read nets and print their symbols if found
# ============================================================

for G in Net.iterator(sys.argv[1]):
    if not G.isLocallyStable():
        print ">>>unstable<<<"
    else:
        found = archive.get(G.minimalImage().systreKey)
        if found:
            print "%s" % found.name
        else:
            print ">>>unknown<<<"

# ============================================================
#   EOF
# ============================================================
